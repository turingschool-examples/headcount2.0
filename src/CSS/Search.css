.search-bar {
  margin: 1rem 0.5rem;
}

.search-input {
  height: 2rem;
  width: 20rem;
  border: 1.5px solid #ccc;
  margin: 1rem 0;
  border-radius: 5px;
}

.search-input:hover {
  box-shadow: 1px 1px 1px 1px #ccc;
}

.hide-suggestions {
  display: none;
}

.did-you-mean {
  font-style: italic;
  color: gray;
}

.show-suggestions {
  margin: 1rem 0;
  text-align: center;
}

.suggestions {
  cursor: pointer;
  border: 1px solid gray;
  margin: 0;
  padding: 2px;
  text-align: center;
}

.suggestions:hover {
  background-color: rgb(244, 243, 243);
  box-shadow: 1.5px 1.5px 1.5px 1.5px #ccc;
}

button {
  width: 21.3rem;
  height: 2rem;
  border-radius: 5px;
  background-color: rgb(230, 230, 230);
  font-size: 1.1rem;
}

button:hover {
  background-color: rgb(142, 142, 142);
  color: rgb(237, 237, 237);
  cursor: pointer;
}

input[type="text"] {
  font-size: 1.1rem;
  line-height: 5rem;
  padding-left: 1rem;
}

.show {
  margin-bottom: 10rem;
  font-style: italic;
}

.show-none {
  display: none;
}

.display-error {
  padding: 1rem 0;
  text-align: center;
}
