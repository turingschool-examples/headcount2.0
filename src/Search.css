.search {
  text-align: center
}

.inputField {
  padding: 10px;
  border: 3px solid red;
  /*margin: 10px;*/
  width: 300px;
}

.inputButton {
  padding: 10px;
  background-color: white;
  margin: 10px;
  border-radius: 30%;
  border: 3px solid green;
  color: green;
}

.inputButton:hover {
  background-color: green;
  color: white;
}
