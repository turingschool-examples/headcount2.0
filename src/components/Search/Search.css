/*SEARCH COMPONENT RELATED CSS GOES HERE*/
.Search {
  width: 35%;
  display: flex;
  justify-content: space-between
}

input {
  height: 2rem;
  width: 100%;
  border: 2px solid #BEBEBE;
  padding-left: .6rem;
  font-size: 1.25rem;
  letter-spacing: 0.5px;
  color: #006989;
}

::placeholder {
  font-family: "Roboto", sans-serif;
  font-weight: 400;
  color: #9E9E9E;
}