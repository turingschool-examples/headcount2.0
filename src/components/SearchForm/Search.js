import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class Search extends Component {
  constructor() {
    super();
    this.state = {
      query: ''
    };
  }

  handleChange =  event => {
    const { value } = event.target;

    this.props.filterDistricts(value);

    this.setState({ query: value });
  }

  render() {
    const { query } = this.state;

    return (
      <input
        onChange={this.handleChange}
        type="text"
        name="query"
        value={query}
        placeholder="Search"
      />
    );
  }
}

Search.propTypes = {
  filterDistricts: PropTypes.func.isRequired
};

export default Search;
