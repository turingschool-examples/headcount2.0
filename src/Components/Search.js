import React, { Component } from "react";
import DistrictRepository from "../helper";
import kindergarners from "../data/kindergartners_in_full_day_program";
import PropTypes from "prop-types";
import "../CSS/Search.css";

export default class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      districtInputOne: "",
      searchSuggestions: []
    };
    this.districtRepository = new DistrictRepository(kindergarners);
  }

  handleChange = ({ value }) => {
    this.setState({ districtInputOne: value });
    this.suggestDistricts();
  };

  suggestDistricts = () => {
    const { districtInputOne } = this.state;
    const searchSuggestions = this.districtRepository.findAllMatches(
      districtInputOne
    );
    this.props.selectedDistricts.forEach(district => {
      searchSuggestions.push(district);
    });

    this.setState({ searchSuggestions });
    if (districtInputOne.length > 1) {
      this.props.handleSubmit(searchSuggestions);
    } else {
      const newDistricts = this.districtRepository.districtsArray();
      this.props.handleSubmit(newDistricts);
    }
  };

  clearInput = () => {
    this.setState({ districtInputOne: "" });
    const newDistricts = this.districtRepository.districtsArray();
    this.props.handleSubmit(newDistricts);
  };

  searchDistrict = district => {
    const filteredDistricts = this.districtRepository.findByName(district);
    this.props.handleSubmit([filteredDistricts]);
  };

  render() {
    return (
      <div className="search-bar">
        <form>
          <input
            type="text"
            className="search-input"
            placeholder={
              this.props.selectedDistricts.length > 0
                ? "Remove Card to Search"
                : "Search District"
            }
            onChange={event => this.handleChange(event.target)}
            value={this.state.districtInputOne}
            disabled={this.props.selectedDistricts.length > 1}
          />
          <button onClick={() => this.props.clearComparisons()}>
            Clear Fields
          </button>
        </form>
        <div
          className={
            this.state.districtInputOne.length > 1
              ? "show-suggestions"
              : "hide-suggestions"
          }
        >
          <span
            className={
              this.state.searchSuggestions.length
                ? "show-none"
                : "display-error"
            }
          >
            None Found! Search Again
          </span>

          <span
            className={
              this.state.searchSuggestions.length ? "show" : "show-none"
            }
          >
            Did you Mean?
          </span>
          {this.state.searchSuggestions.map((district, index) => {
            if (
              this.state.searchSuggestions &&
              this.state.districtInputOne.length > 1
            ) {
              var suggestions = (
                <p
                  className="suggestions"
                  key={index}
                  onClick={event => {
                    this.props.selectCard(event.target.textContent);
                    this.clearInput();
                  }}
                >
                  {district.location}
                </p>
              );
            }
            return suggestions;
          })}
        </div>
      </div>
    );
  }
}

const { func, arrayOf, shape, bool, string, number } = PropTypes;

Search.propTypes = {
  selectedDistricts: arrayOf(
    shape({
      clicked: bool,
      location: string,
      stats: shape({ year: number })
    })
  ),
  handleSubmit: func.isRequired,
  selectCard: func.isRequired,
  clearComparisons: func.isRequired
};
