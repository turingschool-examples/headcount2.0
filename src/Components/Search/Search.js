import React from 'react';
import PropTypes from 'prop-types';
import './Search.css';

const Search = ( {handleSearch} ) => {
  return (
      <div className="search-container">
        <input 
        placeholder="Search for district..."
        className="search"
        onChange={ (e) => handleSearch(e.target.value) }/>
      </div>
    )
}

Search.propTypes = {
  handleSearch: PropTypes.func.isRequired
}

export default Search;