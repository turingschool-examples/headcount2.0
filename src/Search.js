import React, { Component } from 'react';
import './Search.css';
import PropTypes from 'prop-types';

class Search extends Component {
  constructor() {
    super();
    this.state = {
      searchString: ''
    }
  }

  handleImputChange = (e) => {
    const { name, value } = e.target;
    // value = value.toUppercase();
    this.setState( {[name]: value.toUpperCase()})
    this.props.searchForDistrict(value.toUpperCase())
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const searchString = this.state.searchString;
    // console.log(searchForDistrict);
    this.props.searchForDistrict(searchString)
    this.setState({searchString : ''})
  }

  render() {
    return(
      <div className='search'>
        <h3>Enter the district you want to search</h3>
        <form 
          onSubmit={this.handleSubmit}>
          <input 
            className='inputField'
            name='searchString'
            placeholder='Enter District You Want To Search'
            value={this.state.searchString}
            onChange={this.handleImputChange} />
          <button
            className='inputButton'>Submit</button>
        </form>
      </div>
    )
  }
}

Search.propTypes = {
  searchForDistrict: PropTypes.func.isRequired
}

export default Search;