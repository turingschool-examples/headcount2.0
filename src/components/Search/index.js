import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './Search.css'

class Search extends Component {
  constructor({findAllMatches}) {
    super()
    this.state = {
      query: ''
    }
  }

  handleInputChange = event => {
    const { value } = event.target

    this.setState({query: value})
    this.props.findAllMatches(value);
  }

  handleSubmit = event => {
    event.preventDefault()
  }

  render() {
    return(
      <form className="Search" onSubmit={this.handleSubmit} >
        <input
          className="search-input"
          aria-label="Search for a specific district here"
          placeholder="Search for a specific district"
          name="query"
          value={this.state.query}
          onChange={this.handleInputChange}
        />
      </form>
    )
  }
}

Search.propTypes = {
  findAllMatches: PropTypes.func.isRequired
}

export default Search