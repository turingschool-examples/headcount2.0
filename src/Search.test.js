import React from 'react';
import Search from './Search';
import { shallow } from 'enzyme';

describe('Search', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot()
  });


})