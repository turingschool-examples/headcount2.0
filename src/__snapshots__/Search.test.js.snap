// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Enter the district you want to search
</h3>,
        <form
          onSubmit={[Function]}
>
          <input
                    name="searchString"
                    onChange={[Function]}
                    placeholder="Enter District You Want To Search"
                    value=""
          />
          <button>
                    Submit
          </button>
</form>,
      ],
      "className": "search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enter the district you want to search",
        },
        "ref": null,
        "rendered": "Enter the district you want to search",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="searchString"
              onChange={[Function]}
              placeholder="Enter District You Want To Search"
              value=""
/>,
            <button>
              Submit
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "searchString",
              "onChange": [Function],
              "placeholder": "Enter District You Want To Search",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Enter the district you want to search
</h3>,
          <form
            onSubmit={[Function]}
>
            <input
                        name="searchString"
                        onChange={[Function]}
                        placeholder="Enter District You Want To Search"
                        value=""
            />
            <button>
                        Submit
            </button>
</form>,
        ],
        "className": "search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter the district you want to search",
          },
          "ref": null,
          "rendered": "Enter the district you want to search",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="searchString"
                onChange={[Function]}
                placeholder="Enter District You Want To Search"
                value=""
/>,
              <button>
                Submit
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "searchString",
                "onChange": [Function],
                "placeholder": "Enter District You Want To Search",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
